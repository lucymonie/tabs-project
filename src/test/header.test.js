import React from 'react'
import { shallow } from 'enzyme'
import Header from '../components/header'

const component = shallow(<Header />)

it('renders the header component', () => {
  expect(component).toMatchSnapshot()
})

it('renders a header', () => {
  const header = component.find('header')
  expect(header).toHaveLength(1)
})

it('renders some divs', () => {
  const divs = component.find('div')
  expect(divs).toHaveLength(2)
})

it('renders an image', () => {
  const image = component.find('img')
  expect(image).toHaveLength(1)
})
