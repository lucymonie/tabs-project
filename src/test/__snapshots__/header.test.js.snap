// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the header component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row banner"
      >
        <div
          className="logo col s8 offset-s4 m6 offset-m6 l4 offset-l8"
        >
          <img
            alt="The Guardian"
            src="logo.svg"
          />
        </div>
      </div>,
      "className": "col s12 m12 l12",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="logo col s8 offset-s4 m6 offset-m6 l4 offset-l8"
        >
          <img
            alt="The Guardian"
            src="logo.svg"
          />
        </div>,
        "className": "row banner",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="The Guardian"
            src="logo.svg"
          />,
          "className": "logo col s8 offset-s4 m6 offset-m6 l4 offset-l8",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "The Guardian",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row banner"
        >
          <div
            className="logo col s8 offset-s4 m6 offset-m6 l4 offset-l8"
          >
            <img
              alt="The Guardian"
              src="logo.svg"
            />
          </div>
        </div>,
        "className": "col s12 m12 l12",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="logo col s8 offset-s4 m6 offset-m6 l4 offset-l8"
          >
            <img
              alt="The Guardian"
              src="logo.svg"
            />
          </div>,
          "className": "row banner",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="The Guardian"
              src="logo.svg"
            />,
            "className": "logo col s8 offset-s4 m6 offset-m6 l4 offset-l8",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "The Guardian",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
